src/Neo/SmartContract/StorageKey.cs

```diff
@@ -34,7 +34,7 @@ public sealed record StorageKey
 
         public StorageKey() { }
 
-        internal StorageKey(byte[] cache)
+        public StorageKey(byte[] cache)
```

Comment:
**DO NOT** expose the constructor. Their is a better way.

Add this to the `neo.csproj` file to expose the `internal` methods to the different assemblies. If you make this `public` then other developers (non-core devs) will start using it. Than we will have to support this interface.
```xml
<ItemGroup>
  <InternalsVisibleTo Include="Neo.Compiler.CSharp" />
  <InternalsVisibleTo Include="Neo.SmartContract.Testing" />
  <InternalsVisibleTo Include="Neo.SmartContract.Testing.UnitTests" />
</ItemGroup>
```

Reply:
I understand, and the xml code that I posted. Allows those `3` assemblies to use the `internal` methods. You will have access to `ALL` the `internal` methods in `neo.dll`.

Reply:
Both solutions are good for me, changed

Reply:
Just make sure the fully qualified assembly name is correctly spelled. That is what `include` equals.